src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3752,16 +3759,31 @@ private enum PostSkipAction
                     accessorKind = SyntaxKind.UnknownAccessorDeclaration;
                 }
 
-                BlockSyntax body = null;
+                BlockSyntax blockBody = null;
+                ArrowExpressionClauseSyntax expressionBody = null;
+
                 SyntaxToken semicolon = null;
                 bool currentTokenIsSemicolon = this.CurrentToken.Kind == SyntaxKind.SemicolonToken;
-                if (this.CurrentToken.Kind == SyntaxKind.OpenBraceToken || (validAccName && !currentTokenIsSemicolon && !IsTerminator()))
+                bool currentTokenIsArrow = this.CurrentToken.Kind == SyntaxKind.EqualsGreaterThanToken;
+
+                if (this.CurrentToken.Kind == SyntaxKind.OpenBraceToken
+                            || (validAccName && !currentTokenIsSemicolon && !currentTokenIsArrow && !IsTerminator()))
+                {
+                    blockBody = this.ParseBlock(isMethodBody: true, isAccessorBody: true);
```

Comment:
Altering ParseBlockAndExpressionBodiesWithSemicolon was my first attempt, but that didn't work. I will try to change it again, now that I have more insight of the internals.